Gem/Code/Source/Components/BackgroundMusicComponent.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/BackgroundMusicComponent.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <LmbrCentral/Audio/AudioProxyComponentBus.h>
+#include <AzCore/Console/ILogger.h>
+
+namespace MultiplayerSample
+{
+    void BackgroundMusicComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<BackgroundMusicComponent, AZ::Component>()
+                ->Field("Playlist", &BackgroundMusicComponent::m_playlist)
+                ->Version(1);
+
+            AZ::EditContext* editContext = serializeContext->GetEditContext();
+            if (editContext)
+            {
+                editContext->Class<BackgroundMusicComponent>("BackgroundMusicComponent", "Plays a sequence of background music tracks in sequence")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "MultiplayerSample")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &BackgroundMusicComponent::m_playlist, "Playlist", "The set of background music audio triggers to play");
+            }
+        }
+    }
+
+    void BackgroundMusicComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& services)
+    {
+        services.push_back(AZ_CRC_CE("BackgroundMusicComponent"));
+    }
+
+    void BackgroundMusicComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC("AudioProxyService", 0x7da4c79c));
+    }
+
+    void BackgroundMusicComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& services)
+    {
+        services.push_back(AZ_CRC_CE("BackgroundMusicComponent"));
+    }
+
+    void BackgroundMusicComponent::Activate()
+    {
+#if AZ_TRAIT_CLIENT
+        m_audioSystem = AZ::Interface<Audio::IAudioSystem>::Get();
+
+        if (m_audioSystem != nullptr)
+        {
+            m_trackIndex = size_t(-1);
+            ReportTriggerFinished(INVALID_AUDIO_CONTROL_ID);
+        }
+#endif
+
```

Comment:
Suggestion: Consider initializing m_trackIndex to a random number % playlist.size() on each Activate(). Could potentially even make this an optional checkbox on the component. But without it, anyone who tests this game a bunch will hear the first song far more often than the other. ;) 

A slightly bigger variation would be a "shuffle mode" that plays every song in random order exactly once before repeating. But I'd personally be happy with at least "don't always start at the start".

https://github.com/RondeSC/Miller_Shuffle_Algo

Reply:
Yeah, that's a super good call.. that's the exact sort of thing that will drive developers completely nuts.